packages/repo/src/types.ts

```diff
@@ -50,14 +51,14 @@ export type RecordCreateOp = {
   action: WriteOpAction.Create
   collection: string
   rkey: string
-  record: Record<string, unknown>
+  record: RepoRecord
 }
 
 export type RecordUpdateOp = {
   action: WriteOpAction.Update
   collection: string
   rkey: string
-  record: Record<string, unknown>
+  record: RepoRecord
 }
```

Comment:
Re: my last comment on factoring lex away from the repo, this is a useful example of what was on my mind.  `RecordUpdateOp['record']` would be an ipld value, and `updateWriteToOp()` in `packages/pds/src/repo/prepare.ts` would convert the lex record to ipld:
```ts
export const updateWriteToOp = (write: PreparedUpdate): RecordUpdateOp => ({
  action: WriteOpAction.Update,
  collection: write.uri.collection,
  rkey: write.uri.rkey,
  record: lexToIpld(write.record),
})
```

Reply:
Yup that is how I originally had it 👍 

The basic problem is that there is a decent bit of repo calling code where you have to remember to do that serialization. & just sticking it on the entry point to the repo pretty dramatically reduced footguns

Reply:
I'm not opposed to revisiting if you think this is a leaky abstraction. But personally I'm starting to think that we move less in the direction of "repo as grabbag of _any data_ at _any key_" & more so as "repo is lightly lexicon aware, enforces data key path semantics & parseability as lexicon record at the leaves"